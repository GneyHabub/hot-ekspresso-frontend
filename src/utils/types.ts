export type FetchingStatus = "notFetched" | "fetching" | "fetched" | "errorFetching";

export interface Flight {
  from: string;
  to: string;
  departure: Date;
  arrival: Date;
  carrier: string;
}
export interface Booking {
  id: string;
  price: number;
  flights: Flight[];
  serviceClass: ServiceClasses;
}

export interface Airport {
  city: string;
  name: string;
  iata: string;
}

export type ServiceClasses = "Economy" | "Comfort" | "Business";