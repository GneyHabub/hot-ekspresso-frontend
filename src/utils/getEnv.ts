const getEnv = (name: string) => {
  return (import.meta.env[`VITE_${name}`] || "").toString();
}

export default getEnv;