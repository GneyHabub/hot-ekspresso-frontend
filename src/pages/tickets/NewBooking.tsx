import { Search } from '@mui/icons-material';
import { MobileDatePicker } from '@mui/lab';
import { Autocomplete, Box, Button, Card, CardContent, CircularProgress, colors, FormControl, IconButton, MenuItem, Select, Stack, TextField, Typography } from '@mui/material';
import { observer } from 'mobx-react';
import React from 'react';
import store from '../../store/newBookingForm.store';
import getEnv from '../../utils/getEnv';
import { useAuthToken } from '../../utils/hooks/useAuthToken';
import { ServiceClasses } from '../../utils/types';
import BookingItem from './BookingItem';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
import CancelIcon from '@mui/icons-material/Cancel';
import dayjs from 'dayjs';
import PhoneNumberInput from "material-ui-phone-number";
import { useNavigate } from 'react-router';

const NewBooking: React.FC = observer(() => {
  const authToken = useAuthToken({
    domain: getEnv("DOMAIN"),
    audience: "https://dev-od7-l947.us.auth0.com/api/v2/",
    clientId: getEnv("CLIENT_ID"),
    redirect_uri: window.location.origin,
  });
  const navigate = useNavigate();

  React.useEffect(() => {
    if(authToken) {
      store.fetchAirports(authToken);
    }
    return(() => {
      store.resetForm();
    })
  }, [authToken]);

  React.useEffect(() => {
    if(store.submittingStatus === "fetched") {
      navigate("/tickets");
    }
  }, [store.submittingStatus]);

  if(store.airportsfetchingStatus !== "fetched") {
    return <CircularProgress />
  }

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        gap: "15px"
      }}
    >
      <Card>
        <CardContent> 
        <Typography variant="h4" gutterBottom component="div">
          New booking
        </Typography>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'column',
            rowGap: "20px",
            alignItems: "center",
            "& > .form-block": {
              width: "80%"
            }
          }}
        >
          <FormControl
            sx={{
              display: 'flex',
              flexDirection: 'column',
              rowGap: "15px",
              "& > *": {
                display: 'flex',
                justifyContent: "space-evenly",
                gap: "7px"
              }
            }}
            className="form-block"
          >
            <Box>
              <Autocomplete
                freeSolo
                sx={{
                  width: "50%"
                }}
                getOptionLabel={(airport) => `${airport.iata} - ${airport.city}`}
                options={store.airports}
                renderInput={(params) => <TextField required {...params} label="From" />}
                onChange={(e, value) => store.setFromAirport(value)}
              />
              <Autocomplete
                freeSolo
                sx={{
                  width: "50%"
                }}
                getOptionLabel={(airport) => `${airport.iata} - ${airport.city}`}
                options={store.airports}
                renderInput={(params) => <TextField required {...params} label="To" />}
                onChange={(e, value) => store.setToAirport(value)}
              />
            </Box>
            <Box>
              <MobileDatePicker
                label="Date mobile"
                inputFormat="DD/MM/YYYY"
                value={store.departureDate}
                renderInput={(params) => <TextField {...params} sx={{width: "50%"}} label="Departure Date" />}
                onChange={(value) => store.setDepartureDate(value || new Date())}
              />
              <Select
                sx={{
                  width: "50%"
                }}
                value={store.serviceClass}
                onChange={(e) => store.setServiceClass(e.target.value as ServiceClasses)}
              >
                {["Economy", "Comfort", "Business"].map(serviceClass => <MenuItem key={serviceClass} value={serviceClass}>{serviceClass}</MenuItem>)}
              </Select>
            </Box>
          </FormControl>
          <Button
            variant="contained"
            type="submit"
            disabled={!store.formIsValid}
            startIcon={<Search />}
            onClick={() => {
              store.setSelectedFlight(undefined)
              store.searchFlights(authToken)
            }}
          >
            Search flights
          </Button>
        </Box>
        </CardContent>
      </Card>
      {store.flightsFetchingStatus === "fetching" &&
        <CircularProgress />
      }
      {store.flightsFetchingStatus === "fetched" && store.flights.length !== 0 && !store.selectedFlight &&
        <Card>
          <CardContent> 
            {store.flights.map((b, idx) => (
                <BookingItem idx={idx} booking={b} key={b.id} onPick={(booking) => store.setSelectedFlight(booking)}/>
              ))}
          </CardContent>
        </Card>
      }
      {store.selectedFlight &&
        <Card
          raised
          sx={{
            color: "white",
            background: `linear-gradient(145deg, ${colors.blue[500]}, ${colors.blue.A700})`,
            padding: "20px"
          }}
        >
          <Stack
            direction="row"
            justifyContent="space-between"
          >
            <Stack
              direction="row"
              alignItems="center"
              gap="10px"
            >
              <Typography variant="h4">
                {store.selectedFlight.flights[0].from}
              </Typography>
              <ArrowForwardIcon />
              {store.selectedFlight.flights.slice(1).map(flight => (
                <>
                  <Typography variant="h6">
                    {flight.from}
                  </Typography>
                  <ArrowForwardIcon />
                </>
              ))}
              <Typography variant="h4">
                {store.selectedFlight.flights[store.selectedFlight.flights.length - 1].to}
              </Typography>
            </Stack>

            <IconButton
              onClick={() => store.setSelectedFlight(undefined)}
              sx={{
                color: "white"
              }}
            >
              <CancelIcon fontSize="large" />
            </IconButton>
          </Stack>
          <Typography variant="subtitle1">
            {dayjs(store.selectedFlight.flights[0].departure).format("DD-MM-YYYY")}
          </Typography>
        </Card>
      }
      {store.selectedFlight !== undefined &&
        <>
          {store.passengers.map((p, idx) => (
            <Card key={idx}>
              <CardContent>
                <Box
                  sx={{
                    display: "flex",
                    justifyContent: "space-between"
                  }}
                >
                  <Typography variant="h6" gutterBottom>
                    {`Passenger #${idx + 1}`}
                  </Typography>
                  {idx !== 0 &&
                    <IconButton
                      onClick={() => store.deletePassenger(idx)}
                    >
                      <CancelIcon fontSize="medium" />
                    </IconButton>
                  }
                </Box>
                <Box
                  sx={{
                    display: "grid",
                    gridTemplateColumns: "1fr 1fr",
                    gridTemplateRows: "1fr 1fr",
                    gap: "12px",
                    "& .MuiFilledInput-input": {
                      backgroundColor: "white"
                    },
                    "& .phone-input": {
                      backgroundColor: "white",
                      border: "1px solid rgba(0, 0, 0, 0.23)",
                      borderRadius: "4px"
                    }
                  }}
                >
                  <TextField value={p.name} label="Name" onChange={(e) => store.setPassengerName(idx, e.target.value)}/>
                  <TextField value={p.surname} label="Surname" onChange={(e) => store.setPassengerSurname(idx, e.target.value)}/>
                  <TextField value={p.passport} label="Passport number" onChange={(e) => store.setPassengerPassport(idx, e.target.value)}/>
                  <PhoneNumberInput 
                    value={p.phoneNumber} 
                    defaultCountry={'ua'} 
                    variant="filled"
                    inputClass="phone-input"
                    onChange={(value) => {
                    if (typeof value === "string") {
                      store.setPassengerPhoneNumber(idx, value)
                    }
                  }} />
                </Box>
              </CardContent>
            </Card>
          ))}
          <Button
            variant='outlined'
            onClick={() => store.addPassenger()}
            sx={{
              width: "150px"
            }}
          >
            Add passenger
          </Button>
          <Button
            variant='contained'
            onClick={() => store.submit(authToken)}
            sx={{
              width: "300px",
              alignSelf: "center"
            }}
          >
            Reserve
          </Button>
        </>
      }
    </Box>
  )
});

export default NewBooking;