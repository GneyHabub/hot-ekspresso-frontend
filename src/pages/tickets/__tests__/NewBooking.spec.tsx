import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { mocked } from 'ts-jest/utils';
import { LocalizationProvider } from '@mui/lab';
import DateAdapter from '@mui/lab/AdapterDayjs';
import NewBooking from '../NewBooking';

jest.mock('../../../store/newBookingForm.store', () => ({
  submittingStatus: 'fetched',
  airportsfetchingStatus: 'fetched',
  resetForm: jest.fn(),
  fetchAirports: jest.fn(),
}));

describe('NewBooking', () => {
  describe('when authenticated', () => {
    it('matches the snapshot', () => {
      const wrapper = render(
        <LocalizationProvider dateAdapter={DateAdapter}>
          <BrowserRouter>
            <NewBooking />
          </BrowserRouter>
        </LocalizationProvider>,
      );
      expect(wrapper.baseElement).toMatchSnapshot();
    });
  });

  describe('when not authenticated', () => {
    it('matches the snapshot', () => {
      const wrapper = render(
        <LocalizationProvider dateAdapter={DateAdapter}>
          <BrowserRouter>
            <NewBooking />
          </BrowserRouter>
        </LocalizationProvider>,
      );
      expect(wrapper.baseElement).toMatchSnapshot();
    });
  });
});
