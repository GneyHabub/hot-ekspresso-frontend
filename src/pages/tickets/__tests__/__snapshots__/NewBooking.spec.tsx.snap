// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewBooking when authenticated matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1q94rgb"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
          >
            New booking
          </div>
          <div
            class="MuiBox-root css-u6ojzc"
          >
            <div
              class="MuiFormControl-root form-block css-18simf8-MuiFormControl-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-2"
                      id="mui-2-label"
                    >
                      From
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-2"
                        required=""
                        role="combobox"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1ftyaf0"
                        >
                          <span>
                            From
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-4"
                      id="mui-4-label"
                    >
                      To
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-4"
                        required=""
                        role="combobox"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1ftyaf0"
                        >
                          <span>
                            To
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-5"
                    id="mui-5-label"
                  >
                    Departure Date
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Choose date"
                      aria-readonly="true"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-5"
                      readonly=""
                      type="text"
                      value="09/05/2022"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1in441m"
                      >
                        <span>
                          Departure Date
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-rnfpgq-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              Search flights
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`NewBooking when not authenticated matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1q94rgb"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
          >
            New booking
          </div>
          <div
            class="MuiBox-root css-u6ojzc"
          >
            <div
              class="MuiFormControl-root form-block css-18simf8-MuiFormControl-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-8"
                      id="mui-8-label"
                    >
                      From
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-8"
                        required=""
                        role="combobox"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1ftyaf0"
                        >
                          <span>
                            From
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-10"
                      id="mui-10-label"
                    >
                      To
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-10"
                        required=""
                        role="combobox"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-1ftyaf0"
                        >
                          <span>
                            To
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-11"
                    id="mui-11-label"
                  >
                    Departure Date
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Choose date"
                      aria-readonly="true"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-11"
                      readonly=""
                      type="text"
                      value="09/05/2022"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1in441m"
                      >
                        <span>
                          Departure Date
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-rnfpgq-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              Search flights
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
